src/flashdeberta/ops/flash_attention_varlen.py

```diff
@@ -164,42 +224,141 @@ def _fwd_kernel_deberta_disentangled_attention(
+    # If this is some unknown new arch -> default to minimal known for 8th gen
+    elif capability[0] >= 8:
+        max_shared_memory = 99000 - 2000  # Remove 2KB for ops overhead
+    # If this is some older unknown arch -> default to minimal known for < 8th gen
+    else:
+        max_shared_memory = 48000 - 2000  # Remove 2KB for ops overhead
+    
+    # Start with an aggressive configuration
+    if capability[0] >= 8:
         if not causal:
             if D <= 64:
-                BLOCK_M, BLOCK_N, num_stages, num_warps = 128, 64, 3, 4
+                BLOCK_M, BLOCK_N, num_stages, num_warps = 64, 64, 3, 4
             else:
-                if M <= 1024:
-                    BLOCK_M, BLOCK_N, num_stages, num_warps = 128, 32, 3, 4
+                if max_seqlen_q <= 1024:
+                    BLOCK_M, BLOCK_N, num_stages, num_warps = 64, 64, 3, 4
                 else:
-                    BLOCK_M, BLOCK_N, num_stages, num_warps = 128, 128, 3, 8
-        else:
+                    BLOCK_M, BLOCK_N, num_stages, num_warps = 64, 64, 3, 8
+        else:  # causal
             if D <= 64:
-                BLOCK_M, BLOCK_N, num_stages, num_warps = 128, 64, 4, 4
+                if disentangled:
+                    BLOCK_M, BLOCK_N, num_stages, num_warps = 64, 64, 3, 4
+                else:
+                    BLOCK_M, BLOCK_N, num_stages, num_warps = 64, 64, 4, 4
             else:
-                if M <= 1024:
-                    BLOCK_M, BLOCK_N, num_stages, num_warps = 128, 32, 2, 4
+                if max_seqlen_q <= 1024:
+                    BLOCK_M, BLOCK_N, num_stages, num_warps = 64, 64, 2, 4
                 else:
-                    BLOCK_M, BLOCK_N, num_stages, num_warps = 128, 128, 3, 8
-    elif torch.cuda.get_device_capability() == (8, 6):
+                    BLOCK_M, BLOCK_N, num_stages, num_warps = 64, 64, 3, 8
+    elif capability[0] == 7:
         if not causal:
             if D <= 64:
-                BLOCK_M, BLOCK_N, num_stages, num_warps = 128, 64, 3, 4
+                BLOCK_M, BLOCK_N, num_stages, num_warps = 64, 64, 3, 4
             else:
-                BLOCK_M, BLOCK_N, num_stages, num_warps = 128, 32, 2, 4
-        else: # causal
+                BLOCK_M, BLOCK_N, num_stages, num_warps = 64, 64, 2, 4
+        else:  # causal
             if D <= 64:
                 BLOCK_M, BLOCK_N, num_stages, num_warps = 64, 64, 3, 4
             else:
-                BLOCK_M, BLOCK_N, num_stages, num_warps = 128, 32, 2, 4
+                BLOCK_M, BLOCK_N, num_stages, num_warps = 64, 64, 2, 4
     else:
-        BLOCK_M, BLOCK_N, num_stages, num_warps = 32, 32, 1, 4
+        BLOCK_M, BLOCK_N, num_stages, num_warps = 16, 16, 1, 4
+    
+    # Additional adjustments for variable-length sequences
+    
+    # For very sparse batches (many short sequences), reduce block size
+    avg_seq_len = total_tokens / max(1, torch.cuda.device_count())
+    if avg_seq_len < 256:
+        BLOCK_M = min(BLOCK_M, 64)
```

Comment:
```suggestion
        BLOCK_M = min(BLOCK_M, 32)
```

After some further testing it seems that this line was the culprit in making it not working during testing of with short sentences. This transforms the square blocks of 64x64 into a 64x32 block which is causing the rest of the code to complain. Reducing Block size to 32x32 solves the issue.